### DISCUSSION TITLE
blog/cqrs-and-mediatr-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#11] [General]
# blog/cqrs-and-mediatr-in-aspnet-core/

CQRS, or Command Query Responsibility Segregation helps you build super clean and scalable systems with ease. In this article, we will explore this pattern, and use the MediatR package in ASP.NET Core to implement the CQRS Pattern and build a simple yet clean CRUD application in .NET! We will also look into how MediatR notifications work.

https://codewithmukesh.com/blog/cqrs-and-mediatr-in-aspnet-core/

**Comment by shahabfar** [comment]
Thanks for the article.
What if the two notifications has an async handle method?
Should We do any extra other thing?

**Comment by reza250prog** [comment]
Tnx Mukesh . Please create the article you said . Vertical slice + CA + Modular monolith

**Comment by mahendraputra21** [comment]
can't wait the article about Vertical slice + CA + Modular monolith, thanks Mukesh

**Comment by muslumergenc** [comment]
Thanks for the article.
explanatory description

**Comment by raselkobir1** [comment]
Very informatic article. For read and write operation  you use only one data base. I know that it should separate database read/write operation.  after insert data goes to write database and when get any data it goes to read data base, shrink both database.

**Comment by timsar2** [comment]
Hellooo to Mukesh 💯♥️, How to divide Read/Write Database and replicate them?

**Comment by shrinivas-chakravarthy** [comment]
Thanks a lot for the article Mukesh. Please create the article on Vertical slice + CA + Modular monolith. Eagerly waiting to read your article.

**Comment by naufilshaikh09** [comment]
Great article, loving the series so far.

**Reply by sureshramasundaram** [reply]
Awsome article.. your my lead.

**Comment by Sany07** [comment]
If i use CQRS should i have use repository pattern?

**Reply by psdlex** [reply]
for a large scale apps that require lots of tests its better to use them, alot easier to substitute in the tests, for something simple its probably overkill

**Comment by harsh522004** [comment]
Excellent Content Mukesh! You Make it Simple and Interesting.

**Comment by progsherief** [comment]
thanks Mukesh . Please we need the article you said . Vertical slice,
any integration with Azure, 
thx again for your blog
